README.md

```diff
@@ -1,298 +1,433 @@
+functions, and functions that should work with **either** collection type. Your
+job is to develop the code to implement these functions.
 
-## Instructions
+The point of this exercise is to build ***your own implementation*** of the
+collection-processing methods. Don't simply re-use the built-in methods!
+Leverage all you know about callbacks, passing data, etc. to prove that you
+could build your own collection-processing framework whenever ***you*** want.
 
-Your functions should conform to the following guidelines:
+**Hint:** For the first set of functions, you will need to figure out how to
+make them work with either arrays or objects. There are multiple ways you could
+approach this. One option is to use an `if` statement to determine whether the
+collection is an object or an array and then process the collection accordingly.
+However, this approach would require writing two different versions of your code
+for each function, which doesn't sound very efficient. Another (better) option
+is to determine whether the collection is an object or an array and, if it's an
+object, use a JavaScript `Object` method to create an array that contains the
+object's values. You then only need to write code that processes an array,
+regardless of what data structure is passed in to your function. Use your
+Googling skills to figure out how to do this.
 
-1. Write pure functions (see article)
-2. Avoid sharing or mutating state (see article)
-3. Avoid side effects (see article)
+## Collection Functions (Arrays or Objects)
 
-Given the same input your functions should always return the same value.
+### myEach
 
-Below you will find a list of function descriptions detailing what their name,
-parameters and return value should be. Your job is to develop the code to
-implement these functions.
+`myEach(collection, callback)`
 
-The entire `fi` library should be wrapped in an [Immediately Invoked Function
-Expression][IIFE] (IIFE), like the example below.
+Argument(s):
 
-```javascript
-fi = (function() {
-  return {
-    libraryMethod: function() {
-      return "Start by reading the article!";
-    },
-
-    each: function() {
-      /*TODO*/
-    }
-  };
-})();
-
-fi.libraryMethod();
-```
+- a collection (either an object or an array)
+- a callback function
 
-Wrapping a library in code is sometimes called "[The Module Pattern][MP]"
+Return value:
 
-The point of this exercise is to build ***your own implementation*** of the
-collection-processing methods. Don't simply re-use the built-in methods!
-Leverage all you know about callbacks, passing data, etc. to prove that you
-could build your own collection-processing framework whenever ***you*** want.
-
-## Collection Functions (Arrays or Objects)
+- The original collection
 
-**fi.each**
+Behavior:
 
-`fi.each(collection, callback)`
+Iterates over the collection of elements, passing each element in turn to the
+callback function. Returns the original, unmodified, collection for chaining.
```

Comment:
Remove `for chaining` here (and other places it's used) since none of the functions for this exercise can be chained

Reply:
Done